main/main.c

```diff
@@ -25,27 +27,111 @@
 
 #define THRESHOLD_HEIGHT_DIFF 1
 
+// NVS configuration
+#define NVS_NAMESPACE "smartdesk"
+#define NVS_KEY_LOW_HEIGHT "low_height"
+#define NVS_KEY_HIGH_HEIGHT "high_height"
+
 static const char *TAG = "LogicData";
 
 static logicdata_ctx_t *ld = NULL;
 
 static volatile uint8_t height = 0;
 
-static volatile uint8_t low_height = 0;
-static volatile uint8_t high_height = 0;
+static uint8_t low_height = 0;
+static uint8_t high_height = 0;
 
 static volatile bool go_to_height_active = false;
 static volatile uint8_t go_to_height = 0;
 
 static volatile bool btn_up_pressed = false;
 static volatile bool btn_down_pressed = false;
 
+// NVS functions
+static esp_err_t nvs_save_preset_heights(void)
+{
+    nvs_handle_t nvs_handle;
+    esp_err_t err;
+
+    err = nvs_open(NVS_NAMESPACE, NVS_READWRITE, &nvs_handle);
+    if (err != ESP_OK) {
+        ESP_LOGE(TAG, "Error opening NVS handle: %s", esp_err_to_name(err));
+        return err;
+    }
```

Comment:
```suggestion
    RETURN_ON_ERROR(nvs_open(NVS_NAMESPACE, NVS_READWRITE, &nvs_handle));
```

I think this would be nice, then print the final error message from the function.

Reply:
That will only work for this case, in other cases I also need to close the nvs

Reply:
Okay, we can keep this.

Reply:
Updated to use the proposed macro at least in cases where it makes sense.